library/rsa.c

```diff
@@ -1854,7 +1856,7 @@ static int rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
         else
             slen = olen - hlen - 2;
     }
-    else if ( (saltlen < 0) || ((size_t) saltlen > olen - hlen - 2) )
+    else if ( ( (size_t) saltlen ) > olen - hlen - 2 )
```

Comment:
Actually I think the previous version was better: in most builds `RSA_VALIDATE_RET()` does nothing (and actually the option that enables it might be removed in the future), while I think validating `saltlen` should be done in all builds.

Reply:
Done